File: src/main/java/com/thebluehats/server/game/managers/enchants/CustomEnchant.java
package com.thebluehats.server.game.managers.enchants;

import java.util.ArrayList;

import org.bukkit.Material;

public interface CustomEnchant {
    String getName();

    String getEnchantReferenceName();

    ArrayList<String> getDescription(int level);

    boolean isDisabledOnPassiveWorld();

    EnchantGroup getEnchantGroup();

    boolean isRareEnchant();

    Material[] getEnchantItemTypes();
}

File: src/main/kotlin/com/thebluehats/server/game/managers/enchants/CustomEnchant.kt
package com.thebluehats.server.game.managers.enchants

import org.bukkit.Material
import java.util.*

interface CustomEnchant {
    val name: String
    val enchantReferenceName: String
    val isDisabledOnPassiveWorld: Boolean
    val enchantGroup: EnchantGroup
    val isRareEnchant: Boolean
    val enchantItemTypes: Array<Material>
    fun getDescription(level: Int): ArrayList<String>
}